4.assess_fibrosis_reversal/scripts/1.vis_probabilities.r

```diff
@@ -0,0 +1,257 @@
+
+# Filter for Metadata_treatment == "DMSO", then group by Metadata_cell_type and count
+counts_df <- combined_probabilities_df %>%
+  filter(Metadata_treatment == "DMSO") %>%  # Filter for DMSO treatment
+  group_by(Metadata_cell_type) %>%
+  summarise(count = n(), .groups = "drop") %>%  # Drop grouping after summarise
+  ungroup()
+
+# Print counts_df
+head(counts_df)
+
+# Now, split by Metadata_Plate and Metadata_cell_type
+counts_by_plate_df <- combined_probabilities_df %>%
+  filter(Metadata_treatment == "DMSO") %>%  # Filter for DMSO treatment
+  group_by(Metadata_Plate, Metadata_cell_type) %>%
+  summarise(count = n(), .groups = "drop") %>%  # Drop grouping after summarise
+  ungroup()
+
+# Print counts_by_plate_df
+counts_by_plate_df
+
+# Filter for Metadata_treatment == "DMSO", group by Metadata_cell_type and predicted_label, and count
+counts_df <- combined_probabilities_df %>%
+  filter(Metadata_treatment == "DMSO") %>%  # Filter for DMSO treatment
+  group_by(Metadata_cell_type, predicted_label) %>%
+  summarise(count = n(), .groups = "drop") %>%  # Drop grouping after summarise
+  ungroup()
+
+head(counts_df)
+
+# Filter to only include DMSO treatment
+filtered_df <- combined_probabilities_df %>% filter(Metadata_treatment == "DMSO")
+
+height <- 8
+width <- 15
+options(repr.plot.width = width, repr.plot.height = height)
+
+ridge_plot_healthy <- ggplot(filtered_df, aes(x = Healthy_probas, y = Metadata_cell_type, fill = Metadata_treatment)) +
+  geom_density_ridges(aes(fill = Metadata_treatment), alpha = 0.7, scale = 2, rel_min_height = 0.01, bandwidth = 0.1) +
+  scale_fill_manual(values = c("DMSO" = brewer.pal(8, "Dark2")[7])) +  # Only include DMSO color
+  geom_vline(xintercept = 1, linetype = "dashed", color = "black") +
+  scale_x_continuous(breaks = seq(0, 1, 0.5)) +
+  facet_grid(model_type ~ Metadata_Plate, scales = "free_y") +  # Facet by Metadata_Plate instead of Metadata_treatment
+  labs(x = "Probability of healthy prediction", y = "Heart Type") +  # Update x-axis label
+  theme_bw() +
+  theme(
+    legend.position = "none",
+    axis.text = element_text(size = 20),
+    axis.text.x = element_text(size = 20),
+    axis.title = element_text(size = 24),
+    strip.text = element_text(size = 18),
+    strip.background = element_rect(
+      colour = "black",
+      fill = "#fdfff4"
+    )
+  )
+
+# Save figure
+ggsave(paste0("./figures/", batch_var, "_DMSO_controls_ridge_plot.png"), ridge_plot_healthy, height = height, width = width, dpi = 500)
+
+ridge_plot_healthy
+
+# Custom color palette inspired by Dark2
+custom_palette <- c(
+  "#1b9e77",  # Dark2 color 1
+  "#d95f02",  # Dark2 color 2
+  "#7570b3",  # Dark2 color 3
+  "#e7298a",  # Dark2 color 4
+  "#66a61e",  # Dark2 color 5
+  "#e6ab02",  # Dark2 color 6
+  "#a6761d",  # Dark2 color 7
+  "#666666",  # Dark2 color 8
+  "#b3b3b3",  # Light gray (alternative)
+  "#8dd3c7",  # Teal (alternative)
+  "#fb8072"   # Light coral (alternative)
+)
```

Comment:
Consider adding this palette to a R themes file so that it can be imported for other plots too!

Reply:
Fair point! I think this is outside the scope for this PR for now, but I will need to add one once we are making more plots! Thanks!